qiskit_experiments/calibration/calibration_definitions.py

```diff
@@ -0,0 +1,382 @@
+            CalibrationError: If the parameterized channel index is not formatted
+                following index1.index2...
+        """
+        if isinstance(schedules, Schedule):
+            schedules = [schedules]
+
+        for schedule in schedules:
+
+            # check that channels, if parameterized, have the proper name format.
+            #pylint: disable = raise-missing-from
+            for ch in schedule.channels:
+                if isinstance(ch.index, Parameter):
+                    try:
+                        [int(_) for _ in ch.index.name.split('.')]
+                    except ValueError:
+                        raise CalibrationError('Parameterized channel must have a name '
+                                               'formatted following index1.index2...')
+
+            self._schedules[schedule.name] = schedule
+
+            for param in schedule.parameters:
+                if param.name not in self._params:
+                    self._params[param.name] = {}
+
+    def add_parameter_value(self, param: Union[Parameter, str],
+                            value: ParameterValue,
+                            chs: Optional[Union[PulseChannel, List[PulseChannel]]] = None,
+                            ch_type: Type[PulseChannel] = None):
+        """
+        Add a parameter value to the stored parameters. This parameter value may be
+        applied to several channels, for instance, all DRAG pulses may have the same
+        standard deviation. The parameters are stored and identified by name.
+
+        Args:
+            param: The parameter or its name for which to add the measured value.
+            value: The value of the parameter to add.
+            chs: The channel(s) to which the parameter applies. If None is given
+                then the type of channels must by specified.
+            ch_type: This parameter is only used if chs is None. In this case the
+                value of the parameter will be set for all channels of the
+                specified type.
+
+        Raises:
+            CalibrationError: if ch_type is not given when chs are None, if the
+                channel type is not a ControlChannel, DriveChannel, or MeasureChannel, or
+                if the parameter name is not already in self.
+        """
+        if isinstance(param, Parameter):
+            name = param.name
+        else:
+            name = param
+
+        if chs is None:
+            if ch_type is None:
+                raise CalibrationError('Channel type must be given when chs are None.')
+
+            if issubclass(ch_type, ControlChannel):
+                chs = [ch_type(_) for _ in range(self._n_uchannels)]
+            elif issubclass(ch_type, (DriveChannel, MeasureChannel)):
+                chs = [ch_type(_) for _ in range(self._n_qubits)]
+            else:
+                raise CalibrationError('Unrecognised channel type {}.'.format(ch_type))
+
+        if not isinstance(chs, list):
```

Comment:
Do you think tuple and array-like input should be handled as an error?
```suggestion
    try:
        chs = list(chs)
    except TypeError:
        chs = [chs] 
```